src/compiler/checker.ts

```diff
@@ -17866,6 +17866,12 @@ namespace ts {
 
         function symbolHasNonMethodDeclaration(symbol: Symbol) {
             return forEachProperty(symbol, prop => {

```

Comment:
Replacing this function with `!!forEachProperty(symbol, prop => !(prop.flags & SymbolFlags.Method));` seems to pass all tests.

Reply:
Nice! That's a much better solution.